### ISSUE TITLE
Disable quick action dropdown button while running the action

### ISSUE THREAD

**Issue by metas-mk** [metasfresh/metasfresh-webui-frontend#992]
### Type of issue
Feature request
### Current behavior
When starting a quickaction via. Quick Action selector/ dropdown list, the list is not disabled whan a process is running (pending). If the user is fast enough she/ he can open the dropdown during the running process and select another one.
### Expected behavior
Disable the quickaction dropdown button (small square next to quickaction button) when a process is started and pending. This issue is a follow-up issue for https://github.com/metasfresh/metasfresh-webui-frontend/issues/933 (similar solution for green quickaction button).
### Steps to reproduce
Material Receipt Window. In Handling Unit Editor, create a Material Receipt. Then select the receive HU Line, press Quickaction dropdown, select "Korrektur". It runs a time. In that time the Quickaction dropdown shall be disabled.

![image](https://user-images.githubusercontent.com/13365687/27921681-ca34ddc8-6278-11e7-93d5-224724c7d84b.png)

![image](https://user-images.githubusercontent.com/13365687/27921698-dad4d98a-6278-11e7-9c3d-a7791c93a20c.png)

**Comment by teosarca** [comment]
Found following issue: the quick action button remains gray in case i've canceled the process.

Steps to reproduce (that's an example, but happens for other cases too):
* open receipt candidates: https://w101.metasfresh.com:8443/window/540196
* select a row (e.g. first)
* press on the quick action drop down and run Foto
* cancel the process 
** you selected row is still selected => OK
** the quick action button is disabled => NOK - it shall be green again
![image](https://user-images.githubusercontent.com/1244701/28122051-c331b4cc-6725-11e7-96ae-4370ab139d89.png)

**Comment by teosarca** [comment]
update: same thing happens even if the action was executed successfully.

**Comment by metas-rc** [comment]
Results of IT

I found a case where actions stay disabled and they should be enabled. 

* Open picking terminal ( window/540345) 
* For the selected line, press "Pick"
* Open picking slots for the first entry
![image](https://user-images.githubusercontent.com/16001442/28217452-cca1b5f4-68bd-11e7-97d8-21b6faf335db.png)

* Press Pick HU:
![image](https://user-images.githubusercontent.com/16001442/28217478-df21fca2-68bd-11e7-8334-af157187f6f3.png)


![image](https://user-images.githubusercontent.com/16001442/28217517-fd59445a-68bd-11e7-8ddf-815e549a343a.png)


* Press Cancel
![image](https://user-images.githubusercontent.com/16001442/28217540-1d292264-68be-11e7-9b60-af41de6f7fcc.png)

* => NOK: The actions are all disabled. They should have been recalculated, I think. 
* Note; whe n I pressed Cancel there was nothing new in inspection

**Comment by wiadev** [comment]
Root cause of the issue:
*  includedView drops selection when action invoked using dropdown menu
*  quickActions also lacked of additional checks to cover opening/closing of second layer modals

**Comment by metas-ts** [comment]
cherry-icked the cahnge into the `release` branch (https://jenkins.metasfresh.com/job/ops/job/cherry-pick-issue-script/200/)

**Comment by metas-rc** [comment]
Results of IT 2

Tested on webui

OK: Pressing cancel after starting a process works fine now. The actions are available again for the selected entry.

NOK : The initial problem was not solved. I can start a process from dropdown, and while that process is running the dropdown is still available and I can start other process.

**Comment by teosarca** [comment]
I think the scenario, how it happens now is:
1. user clicks on quick action button
2. frontend disables the button 
3. frontend starts the process/action. The process/action returns and tells the frontend to open a modal view
4. frontend re-enables the button because the process was executed
5. frontend opens and displays the modal

Loud thinking, i believe what we need to change is something like:
(note: below is iterative pseudocode... you will have to "translate" it to reactive)

```
disable_button();
try
{
   result = execute_process_and_get_result();
   check result and open modal or whatever is needed;
}
catch (exception)
{
    handle the exception;
}
finally
{
    enable_button();
}
```

Basically, executing the process and performing the follow-up actions (like open the modal) shall be part of the "execution" block.

wdyt?

**Comment by teosarca** [comment]
Checked PR #1069 and found following issues:

1.  when trying to receive an handling unit (HU) from receipt schedule, on HU editor (modal) the quick actions are wrong (for selected line).
Steps to reproduce:
* open receipt schedules, e.g. https://w101.metasfresh.com:8443/window/540196
* select a line where u can receive (i.e. the quick action button will show something like "4 G1 x 10Kg"
* click on it => the HU editor modal is displayed, the first line is selected but the quickActions are wrong.
![image](https://user-images.githubusercontent.com/1244701/28523636-f9e7986a-7085-11e7-9b24-2401b13f3bd3.png)





2.  The quick actions button does not stay disabled while the process is running
![image](https://user-images.githubusercontent.com/1244701/28523674-20361f78-7086-11e7-9940-41b8e4afcf83.png)

**Comment by teosarca** [comment]
Found another issue:
![image](https://user-images.githubusercontent.com/1244701/28562377-9e6ef492-712a-11e7-8565-f84015e5ee80.png)

----


Steps to reproduce are:
* open manufacturing orders: https://w101.metasfresh.com:8443/window/53009
* select a row where you can "Issue/Receipt", e.g.
![image](https://user-images.githubusercontent.com/1244701/28562422-d23d0ae8-712a-11e7-8c28-364bdba455f2.png)

* click the Issue/Receipt quick actions so the modal will popup
* select a row of type "CO" (component) and press Issue HUs, .e.g.
![image](https://user-images.githubusercontent.com/1244701/28562490-15a68782-712b-11e7-941f-841c61e92dac.png)

* on the right side an included view shall be opened showing the HUs which are available to issue
* also the right side included view has it's own "quickActions" button => that one is not populated correctly.

NOTE: this one is working on master but not on this branch.

**Comment by teosarca** [comment]
Found another issue: when a websocket event is received by frontend, the frontend is refreshing the data from view => OK
But, after refreshing the data it shall also re-query the quick actions for current selected row because it might be that the row changed so the quick actions changed to.

----

Steps to reproduce:
* open receipt schedules: https://w101.metasfresh.com:8443/window/540196
* select a line and press the quick action to receive an HU, e.g.
![image](https://user-images.githubusercontent.com/1244701/28562924-cab086f4-712c-11e7-8f51-3edf1fbf074d.png)

* the HU editor modal opens, select the top line and call "Create material receipt" quick action
* after the material receipt is created, the backend will send a websocket event to inform the frontend that it shall refresh the view content
* the frontend is doing this (remark that the HU turned to gray) but it's not refreshing the quick actions

NOTE: this one is working on master but not on this branch.

**Comment by wiadev** [comment]
Test cases I've tried:

* Open window/540196
* Click on any line with "Packings Description" looking like "4 G1 x 10"
* Click green action button stating something like "4 G1 x 10"
* Make sure green action button has "Create material receipt"
* Click the button
* Button should stay gray until process complete
* Then button title should be changed to "Dispose" and become green

***

* Open window/540196
* Click on any line with "Packings Description" looking like "4 G1 x 10"
* Click green action button stating something like "4 G1 x 10"
* Click on any line which has HUCode of empty value
* Make sure green action button has "Dispose" value
* Click the button
* Button should stay gray until process throws the error
* Then action button should become green

***

* Open window/540196
* Click on any line
* Click down arrow next to green action button and pick "Foto"
* Modal appears
* Click "Cancel"
* Document list appears again
* Action button should become green and contain the same action it had before

**Comment by wiadev** [comment]
@teosarca the above are the test cases I've tried, FYI.

**Comment by teosarca** [comment]
Checked PR #1085 and works nice and smooth... but i will not integrate because i've found issues on the other task which is part of that PR.

**Comment by metas-lc** [comment]
IT
1. cannot use quickation dropdown while a process is running OK
2. the quick action button doesn't remain gray in case i've canceled a process. OK
3. same like case 2 when pressing done to a process OK
4. when opening a process (e.g. one to receive) the quickactions are correct OK
5. in issue/receipt, the quickactions for the right side are called using the selected row from the left side OK
